src/shared/database/amendments.ts

```diff
@@ -9,78 +13,142 @@ import { query } from './utils'
+  'PayChan',
+  'CryptoConditions',
+  'TickSize',
+  'fix1368',
+  'Escrow',
+  'fix1373',
+  'EnforceInvariants',
+  'SortedDirectories',
+  'fix1201',
+  'fix1512',
+  'fix1523',
+  'fix1528',
+]
 
 const AMENDMENT_VERSION_REGEX =
   /\| \[(?<amendmentName>[a-zA-Z0-9_]+)\][^\n]+\| (?<version>v[0-9]*\.[0-9]*\.[0-9]*|TBD) *\|/u
 
-// TODO: Clean this up when this PR is merged:
-// https://github.com/XRPLF/rippled/pull/4781
+export const NETWORKS_HOSTS = new Map([
+  ['main', 'ws://s2.ripple.com:51233'],
+  ['test', 'wss://s.altnet.rippletest.net:51233'],
+  ['dev', 'wss://s.devnet.rippletest.net:51233'],
+])
+
 /**
- * Fetch a list of amendments names from rippled file.
+ * Fetch amendments information including id, name, and deprecated status.
  *
- * @returns The list of amendment names.
+ * @returns Void.
  */
-async function fetchAmendmentNames(): Promise<Map<string, boolean> | null> {
-  try {
-    const response = await axios.get(
-      'https://raw.githubusercontent.com/XRPLF/rippled/develop/src/libxrpl/protocol/Feature.cpp',
-    )
-    const text = response.data
-    const amendmentNames: Map<string, boolean> = new Map()
-    text.split('\n').forEach((line: string) => {
-      const name = ACTIVE_AMENDMENT_REGEX.exec(line)
-      if (name) {
-        amendmentNames.set(name[1], name[0].includes('VoteBehavior::Obsolete'))
-      } else {
-        const name2 = RETIRED_AMENDMENT_REGEX.exec(line)
-        if (name2) {
-          amendmentNames.set(name2[1], true)
-        }
-      }
-    })
-    return amendmentNames
-  } catch (err) {
-    log.error('Error getting amendment names', err)
-    return null
+async function fetchAmendmentsList(): Promise<void> {
+  for (const [network, url] of NETWORKS_HOSTS) {
+    await fetchNetworkAmendments(network, url)
   }
 }
 
 /**
- * Extracts Amendment ID from Amendment name inside a buffer.
+ * Fetch amendments information including id, name, and deprecated status of a network.
  *
- * @param buffer -- The buffer containing the amendment name.
+ * @param network - The network being retrieved.
+ * @param url - The Faucet URL of the network.
  *
- * @returns The amendment ID string.
+ * @returns Void.
  */
-function sha512Half(buffer: Buffer): string {
-  return createHash('sha512')
-    .update(buffer)
-    .digest('hex')
-    .toUpperCase()
-    .slice(0, 64)
+async function fetchNetworkAmendments(
+  network: string,
+  url: string,
+): Promise<void> {
+  try {
+    log.info(`Updating amendment info for ${network}...`)
+    const client = new Client(url)
+    await client.connect()
+    const featureAllResponse: FeatureAllResponse = await client.request({
+      command: 'feature',
+    })
+
+    const featuresAll = featureAllResponse.result.features
+
+    for (const id of Object.keys(featuresAll)) {
+      amendmentIDs.set(id, {
+        name: featuresAll[id].name,
+        deprecated: RETIRED_AMENDMENTS.includes(featuresAll[id].name),
+      })
+      votingAmendments.delete(id)
```

Comment:
Why are we removing amendments from the voting list?

Reply:
nvm, understand

Reply:
If an amendment info has been retrieved for amendments in voting I will just remove it so the next loop will not have to look up that info again

Reply:
Just renamed the variable to avoid confusion